### ISSUE TITLE
[Help Wanted] Is it possible to authenticate with Azure CLI using service principal

### ISSUE THREAD

**Issue by jdneo** [Azure/azure-libraries-for-java#417]
I met an issue that cannot auth with Azure CLI using Service Principal, the steps are:

1. login to azure through service principal (https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli?view=azure-cli-latest#logging-in-with-a-service-principal)
2. Get Authenticated Object through `azureConfigure().authenticate(AzureCliCredentials.create())`

And then when I interactive with Azure, I will got `clientId is null or empty` error.

Is it because when we authenticate with Service Principal in Azure CLI, an entry will be generate in the `accessTokens.json` which has no `_clientId` field? The format for that entry is:

```json
{
  "servicePrincipalId": "---",
  "servicePrincipalTenant": "---",
  "accessToken": "---"
}
```

**Comment by jdneo** [comment]
@jianghaolu Any feedback about this issue?

**Comment by jdneo** [comment]
Do we have any updates for this?

**Comment by yugangw-msft** [comment]
@jeneo, I am pretty sure `servicePrincipalId` is `clientId`. No familiar with Java SDK, but it should either read 
 `user.name` from azureProfile.json or read `servicePrincipalId` from `accessTokens.json`

**Comment by xccc-msft** [comment]
Hi @jdneo the service principal id is stored in AzureCliCredentials. We will plan to expose it.

**Comment by xccc-msft** [comment]
As discussed with @Flanker32 , the issue got resolved by update client-runtime to 1.6.15. Close this now.
